Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveyException.kt

```diff
@@ -9,11 +9,13 @@ class SurveyException constructor(
 ) : Exception("$type: $message", cause) {
 
     enum class Type {
-        ALREADY_PARTICIPATED_THIS_MONTH
+        ALREADY_PARTICIPATED_THIS_MONTH,
+        OTP_NOT_AUTHORIZED
     }
 }
 
 fun SurveyException.errorMsgRes(): Int = when (type) {
     SurveyException.Type.ALREADY_PARTICIPATED_THIS_MONTH ->
         R.string.datadonation_details_survey_consent_error_ALREADY_PARTICIPATED
+    else -> R.string.datadonation_details_survey_consent_error_TRY_AGAIN_LATER
```

Comment:
Don't add else here, because we want to be reminded to handle new error types

Reply:
changed as ordered